lib/elliptic/eddsa/index.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+var bn = require('bn.js');
+var hash = require('hash.js');
+var elliptic = require('../../elliptic');
+var utils = elliptic.utils;
+var assert = utils.assert;
+
+function parseBytes(bytes) {
+  return typeof bytes === 'string' ? utils.toArray(bytes, 'hex') :
+                                     bytes;
+}
+
+function EDDSA(curve) {
+  if (!(this instanceof EDDSA))
+    return new EDDSA(curve);
+
+  assert(typeof curve === 'string');
+  assert(curve === 'ed25519', 'only tested with ed25519 so far');
+  var curve = this.curve = elliptic.curves[curve].curve;
+  this.g = curve.g;
+  this.g.precompute(curve.n.bitLength() + 1);
+  this.encBytes = Math.ceil(curve.n.bitLength() / 8);
+  this.hash = hash.sha512;
+}
+
+module.exports = EDDSA;
+
+/**
+* @param {Array} message - message bytes
+* @param {Array|String} seed - secret seed bytes
+* @returns {bytes} - signature
+*/
+EDDSA.prototype.sign = function(message, secret) {
```

Comment:
May I ask you to avoid anonymous functions? `function sign(message, secret) {`... ;)